### PR TITLE
update list of programs to nauty 2.8.8

### PR DESCRIPTION
**Pull request by dcoudert**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

This PR updates the list of programs of nauty to the list of version 2.8.8. For instance, genktreeg has been introduced in version 2.8.8 and we want to use it in https://github.com/sagemath/sage/pull/36924

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -25,9 +25,9 @@ autoreconf -vfi
 ./configure --disable-popcnt --disable-clz --enable-generic --enable-tls
 make
 
-programs="addedgeg amtog biplabg catg complg converseg copyg cubhamg deledgeg delptg directg dreadnaut dretodot dretog \
-  genbg genbgL geng genquarticg genrang genspecialg gentourng gentreeg hamheuristic labelg linegraphg listg multig newedgeg \
-  planarg ranlabg showg subdivideg twohamg vcolg watercluster2 NRswitchg"
+programs="addedgeg amtog biplabg catg complg converseg copyg countneg cubhamg deledgeg delptg directg dreadnaut dretodot dretog \
+  genbg genbgL geng genktreeg genquarticg genrang genspecialg gentourng gentreeg hamheuristic labelg linegraphg listg multig newedgeg \
```
Comment (mkoeppe) [inline recipe/build.sh:29]: More programs are missing that were added in 2.8.6 or earlier. Compare with the list in https://github.com/sagemath/sage/pull/36924#issuecomment-1866456706

Comment (dcoudert) [reply recipe/build.sh]: Should be ok with last commit (same list than in Sagemath).

**Comment by isuruf** [comment]
Thanks

**Comment by dcoudert** [comment]
I realize that it might not be a good idea to ad all programs of nauty in the list of programs due to the following test:
```
if [[ "$target_platform" != win-* ]]; then
    programs="$programs countg pickg shortg"
fi
```

**Comment by isuruf** [comment]
There were compilation errors for those programs in windows, but we've disabled windows for now anyways.

**Pull request merged**